src/main/scala/beam/agentsim/agents/vehicles/BeamVehicle.scala

```diff
@@ -102,7 +102,11 @@ class BeamVehicle(
     val distanceInMeters = beamLeg.travelPath.distanceInM
     val network = beamServices.matsimServices.getScenario.getNetwork
     val fuelConsumption = BeamVehicle.collectFuelConsumptionData(beamLeg, beamVehicleType, beamServices.networkHelper)
-    val energyConsumed = powerTrain.estimateConsumptionInJoules(fuelConsumption)
+    val energyConsumed =
+      beamServices.vehicleEnergy.getFuelConsumptionEnergyInJoulesUsing(
+        fuelConsumption,
+        overallFallBack = powerTrain.estimateConsumptionInJoules,
+        singleFallBack = powerTrain.estimateConsumptionInJoules)
```

Comment:
Not clear what "overal" vs "single" means. Also, see below, these are return an energy estimate but you are using that as if it were a rate.

Reply:
Yah, overall vs single is if the entire thing results in a None - maybe that isn't a realistic case though. Whereas single is if a single rate lookup yields a None. Typing that out, I guess that doesn't make sense since the single fallback will always yield something. I'll remove the overall then.